trin-history/src/utils.rs

```diff
@@ -0,0 +1,93 @@
+use trin_core::{
+  portalnet::{
+    U256,
+    types::{HistoryContentKey, HeaderKey, BodyKey, ReceiptsKey},
+  },
+  utils::content_key::{keccak}
+};
+use hex;
+
+// content-id derivation function for history network
+pub fn history_network_derive_content_id(key: Vec<u8>) -> Result<U256, String> {
```

Comment:
I'm thinking that these should be methods of the `ContentKey` enums defined in `types.rs` instead of external `util` functions, and then I'll update`PortalStorage` methods take a `ContentKey` and just call `derive_content_id` on it internally instead of having to pass in the function like we're doing now. That will be simpler and better organized.

Any reason you can see that we shouldn't do that?

Reply:
This means you can get rid of the closure function in PortalStorage right? Yeah 100% I think this has multiple benefits. The only thing is, since it is now a method, we have to call it like `key.derive_content_id()`, where `key` is of type `ContentKey`, as I understand. So we'd need to get the type first, which means instead of passing a byte array to `derive_content_id` and then decoding in that method, we would have to decode the byte array first somewhere outside of the method, which gives us the type, and then we can call the derivation function. But regardless, I think this way of doing things is much cleaner and organized.

Reply:
👍 I'd think you can just do a `match` on the first byte inside the function but I might not be seeing what you mean fully.

Either way, let's go that route. Once you merge this in I'll refactor PortalStorage to take the `ContentKey` type you define in this PR.